subt/artf_node.py

```diff
@@ -133,13 +162,65 @@ def detect(self, image):
         result = self.detector(img)
         result_cv = self.cv_detector(img)
         if result or result_cv:
-            self.stdout(result, result_cv)  # publish the results independent to detection validity
+            # publish the results independent to detection validity
+            self.stdout(result, result_cv)  # for debugging in case of crash
+            self.publish('debug_result', str(result))
+            self.publish('debug_cv_result', str(result_cv))
             checked_result = check_results(result, result_cv)
             if checked_result:
-                report = result2report(checked_result, self.depth)
+                report = result2report(checked_result, self.depth, self.fx)
                 if report is not None:
                     self.publish('artf', report)
                     self.publish('debug_artf', image)  # JPEG
                     self.publish('debug_depth', self.depth)
 
+
+if __name__ == "__main__":
+    # run specific test based on recorded stdout
+    import argparse
+    from datetime import timedelta
+    from osgar.lib.serialize import deserialize
+    from osgar.logger import LogReader, lookup_stream_id, lookup_stream_names
+    from ast import literal_eval
+
+    parser = argparse.ArgumentParser(description='Test 3D reports')
+    parser.add_argument('logfile', help='OSGAR logfile')
+    parser.add_argument('--time-limit-sec', '-t', help='cut time in seconds', type=float)
+    args = parser.parse_args()
+
+    names = lookup_stream_names(args.logfile)
+    stdout_stream_id = lookup_stream_id(args.logfile, 'detector.stdout')
+    streams = [stdout_stream_id]
+    if 'detector.debug_depth' in names:
+        depth_stream_id = lookup_stream_id(args.logfile, 'detector.debug_depth')
+        streams.append(depth_stream_id)
+    else:
+        depth_stream_id = None
+
+    # dummy depth for loder logfiles
```

Comment:
s/loder/older ?

Reply:
@jisa thanks for comments/review - this part is still "draft" and I was thinking to drop it (as I was unable to debug the logging - I am no longer able to run simulation "reasonably", and loop via CloudSim with broken logging is not nice either :( 
Now my plan is to support only new logs and ignore old ones, i.e. this section I wil rework now.